.jenkins/library/vars/helpers.groovy

```diff
@@ -0,0 +1,317 @@
+                    mkdir build
+                    cd build
+                    ${ninjaBuildCommand(cmakeArgs, "..")}
+                    ninja run || exit !ERRORLEVEL!
+                )
+            """
+        )
+    }
+}
+
+/**
+ * Builds Open Enclave using cmake and Ninja.
+ *
+ * @param cmake_arguments String of arguments to be passed to cmake
+ * @param build_dir       String that is a path to the directory that contains CMakeList.txt
+ *                        Can be relative to current working directory or an absolute path
+ */
+def ninjaBuildCommand(String cmake_arguments = "", String build_directory = "${WORKSPACE}") {
+    if(isUnix()) {
+        return """
+            set -x
+            cmake ${build_directory} ${cmake_arguments}
+            ninja -v
+        """
+    } else {
+        return """
+            @echo on
+            setlocal EnableDelayedExpansion
+            cmake ${build_directory} ${cmake_arguments} || exit !ERRORLEVEL!
+            ninja -v || exit !ERRORLEVEL!
+        """
+    }
+}
+
+/**
+ * Downloads an Open Enclave release version from a pre-defined Azure Blob container and installs it.
+ * Warning: this function (and any others that runs non-shell commands) must not be called 
+ *          within a shell otherwise a null command would be ran after this function completes.
+ *
+ * @param release_version  The version of the Open Enclave release to install
+ * @param oe_package       Open Enclave package to install
+ *                         - "open-enclave" [Default]
+ *                         - "open-enclave-hostverify"
+ * @param source           Which source to download Open Enclave from
+ *                         - "Azure" to download from the Azure blob storage [Default]
+ *                         - "GitHub" to download from the Open Enclave Repository
+ */
+def releaseInstallCommand(String release_version = null, String oe_package = "open-enclave", String source = "Azure") {
+    // Check parameters are valid
+    if(!release_version) {
+        error("[Error] Invalid Open Enclave release version defined!")
+    }
+    if(!["open-enclave", "open-enclave-hostverify"].contains(oe_package)) {
+        error("[Error] Invalid Open Enclave package defined!")
+    }
+    if(!["Azure", "GitHub"].contains(source)) {
+        error("[Error] Invalid Open Enclave source defined!")
+    }
+    // For *nix
+    if(isUnix()) {
+        // Determine distribution and version
+        // Note: lsb_release is only available on Ubuntu.
+        def os_id = sh(
+            script: "lsb_release --id --short",
+            returnStdout: true
+            ).trim()
+        def os_release = sh(
+            script: "lsb_release --release --short | sed 's/\\.//'",
+            returnStdout: true
+            ).trim()
+        if(source == "Azure") {
+            // Download from Open Enclave storage container
+            azureDownload(
```

Comment:
What are your thoughts on having wrapper functions for each download source, ie. `downloadAzureRelease()`/`downloadGithubRelease()`?

Reply:
Added `azureContainerDownload`, `releaseDownloadLinuxGitHub`, and `releaseDownloadWindowsGitHub`